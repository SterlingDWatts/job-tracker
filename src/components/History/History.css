.History {
  padding-top: 16px;
}
.History li {
  display: grid;
  grid-template-columns: 120px auto;
  font-size: 14px;
}
