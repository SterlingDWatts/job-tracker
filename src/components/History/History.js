import React from "react";
import { format } from "date-fns";
import "./History.css";

export default function History(props) {
  let { column, columns } = props;
  column = column.map((col, idx) => {
    let colName = columns.find((el) => el.id === col.id);
    colName = colName.name;
    return (
      <li key={idx}>
        <div className="name">{colName}</div>
        <div className="date">{format(col.date, "MMM dd")}</div> 
      </li>
    );
  });
  return (
    <div className="History">
      <ul>{column}</ul>
    </div>
  );
}
