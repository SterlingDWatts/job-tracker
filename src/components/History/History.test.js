import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { BrowserRouter } from "react-router-dom";
import History from "./History";

describe("History Component", () => {
  const columns = [
    { id: 0, name: "Applied" },
    { id: 1, name: "Intro Interview" },
    { id: 2, name: "Recruiter Call" },
    { id: 3, name: "Sent Follow-up" },
    { id: 4, name: "Coding Challenge" },
    { id: 5, name: "Sent Follow-up" },
    { id: 6, name: "Technical Interview" },
    { id: 7, name: "Sent Follow-up" },
    { id: 8, name: "Onsite Interview" },
    { id: 9, name: "Offer Letter Date" },
  ];
  const column = [
    { id: 0, date: new Date(2020, 4, 14) },
    { id: 4, date: new Date(2020, 4, 14) },
  ];
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <History columns={columns} column={column} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the UI as expected", () => {
    const wrapper = shallow(<History column={column} columns={columns} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
