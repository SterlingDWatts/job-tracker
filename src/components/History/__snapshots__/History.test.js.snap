// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History Component renders the UI as expected 1`] = `
<div
  className="History"
>
  <ul>
    <li
      key="0"
    >
      <div
        className="name"
      >
        Applied
      </div>
      <div
        className="date"
      >
        May 14
      </div>
    </li>
    <li
      key="1"
    >
      <div
        className="name"
      >
        Coding Challenge
      </div>
      <div
        className="date"
      >
        May 14
      </div>
    </li>
  </ul>
</div>
`;
